form-los-altos-da.html

```diff
@@ -38,9 +38,9 @@
                      class="no-load-init init-on-submit"
                      value="Damage Summary for BAT(s) {{selected-fields:[name$='batnum']|remove:|sort:num|join:, }}"
                      required="required"
-                     pattern="Damage Summary for BAT\(s\) .*"
+                     pattern="Damage Summary for BAT\(s\).*"
                      title="This field must have a prefix of
-                            &quot;Damage Summary for BAT(s) &quot;"/>
+                            &quot;Damage Summary for BAT(s)&quot;"/>
```

Comment:
I'm not sure what that means. I've seen the form be not submittable because there were no details and consequently no trailing space. Do you think the trailing space is required, for the message to be valid?

Reply:
The template that auto-generates the subject always includes a space, so this should only have happened if you manually changed the subject. Currently the form does allow the subject to change although it is deliberately styled to discourage that because having a consistent format for the subject that indicates what data is available in the message should facilitate manual message processing on the receiver side. Is there an important use case where you imagine the subject needs to be changed?

Reply:
I didn't edit the subject, and I don't know why an operator would edit the subject.

I encountered the problem when I created a Damage Assessment, submitted it and then opened it. I created it with all empty fields in the big table; that is no BAT reports. So, {{selected-fields:[name$='batnum']|remove:|sort:num|join:, }} evaluated to an empty string, and write_pacforms_representation called bracket_data("Damage Summary for BAT(s) ") with a trailing space. But bracket_data calls trim(), which removes the trailing space, resulting in "10.: [Damage Summary for BAT(s)]". I submitted the message to Outpost without difficulty, but when I opened the same message, the form's Subject field was invalid because it had no trailing space.

Either the validation pattern should be relaxed or bracket_data should not trim().

Reply:
Ah. That makes sense. I think the right thing to do here is probably to tweak validation so that the form can't be submitted without at least one BAT report. A regular ICS213 should be used for messages that aren't reporting BAT status. The simplest solution might just be to require a number in the BAT field of the first line of the status matrix.